src/utils/fetchErrors.js

```diff
@@ -0,0 +1,20 @@
+class ServerError extends Error {
+}
+
+const throwWithEndUserMessage = (response, json, friendlyMessage) => {
+  if (!response.ok) {
+    let messageToSend = json.message;
+    if (response.status === 500) {
+      console.error(`Status 500 fetching ${response.url}. Hidding full error from user: ${messageToSend}`);
```

Comment:
I have removed the `JSON.stringify()` from your proposal, since `message` should be a plain string. I have left it as
```
      console.error(`Status 500 fetching ${response.url}. Error in response: ${message}`);
```